workflow/controller/entrypoint/container_registery_index.go

```diff
@@ -0,0 +1,53 @@
+package entrypoint
+
+import (
+	"context"
+
+	"github.com/google/go-containerregistry/pkg/authn/k8schain"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type containerRegistryIndex struct {
+	kubernetesClient kubernetes.Interface
+}
+
+func (i *containerRegistryIndex) Lookup(ctx context.Context, image string, options Options) (*Image, error) {
+	kc, err := k8schain.New(ctx, i.kubernetesClient, k8schain.Options{
+		Namespace:          options.Namespace,
+		ServiceAccountName: options.ServiceAccountName,
+		ImagePullSecrets:   imagePullSecretNames(options.ImagePullSecrets),
+	})
+	if err != nil {
+		return nil, err
+	}
+	ref, err := name.ParseReference(image)
+	if err != nil {
+		return nil, err
+	}
+	img, err := remote.Image(ref, remote.WithAuthFromKeychain(kc))
+	if err != nil {
+		return nil, err
+	}
+	f, err := img.ConfigFile()
+	if err != nil {
+		return nil, err
+	}
+	cmd := &Image{}
+	if len(f.Config.Entrypoint) > 0 {
+		cmd.Cmd = f.Config.Entrypoint
+	} else {
+		cmd.Cmd = f.Config.Cmd
+	}
```

Comment:
Just noticed this, this still seems wrong.  Shouldn't it be:

```
	cmd := &Image{}
	if len(f.Config.Entrypoint) > 0 {
		cmd.Entrypoint = f.Config.Entrypoint
	}
	if len(f.Config.Cmd) > 0 {
		cmd.Cmd = f.Config.Cmd
	}
```

though i guess both if's are unnecessary